File: content-solutions/queue-deque/queue-hot-potato.js
function solution(n, k) {
  const circle = Array.from({ length: n }, (_, i) => i + 1);
  let head = 0;
  const out = [];
  while (circle.length - head > 0) {
    for (let i = 0; i < k - 1; i += 1) circle.push(circle[head++]);
    out.push(circle[head++]);
  }
  return out;
}

File: content-solutions/queue-deque/queue-hot-potato.java
import java.util.*;

class Solution {
    public List<Integer> solution(int n, int k) {
        Queue<Integer> circle = new ArrayDeque<>();
        for (int i = 1; i <= n; i++) circle.offer(i);
        List<Integer> out = new ArrayList<>();
        while (!circle.isEmpty()) {
            for (int i = 0; i < k - 1; i++) circle.offer(circle.poll());
            out.add(circle.poll());
        }
        return out;
    }
}
